hack/verify-pylint.sh

```diff
@@ -38,4 +38,4 @@ shopt -s extglob globstar
 
 # TODO(clarketm) there is no version of `pylint` that supports "both" PY2 and PY3
 # I am disabling pylint checks for python3 files until migration complete
```

Comment:
Does something prevent us from having both py2 and py3 builds of pylint and using the correct linter for each module?

Reply:
@fejta - my initial thought was exactly this – to specify a separate set of requirements/dependencies (e.g. `requirements.txt` / `requirements3.txt`) for py2 and py3. This would be necessary for `pylint`,  since the other external dependencies, there is no one version that is compatible with both py2 and py3. 

Upon implementation, I ran into complications with achieving this build concurrency due to limitations with **[bazelbuild/rules_python](https://github.com/bazelbuild/rules_python)** [pip_import](https://github.com/bazelbuild/rules_python#importing-pip-dependencies) directive that we are using. While there are [PRs](https://github.com/bazelbuild/rules_python/pull/179/commits/23b1a6ee0d1252408318da376f69847cabfeedb5) in the works to circumvent this issue, currently there is not a mechanism to specify the *version* of python interpreter, rather `pip_import` uses the *system* python interpreter.

With additional work this may be possible; otherwise, my though was the lint before checkin, then make the switch as the end and likely provide additional documentation on this.

Reply:
Is there an issue in the rules_python repo that describes the problem? If not can we make one?

Reply:
There are several issues that reference to this problem in one form or another. The most general, and recent, of which that I came across is [here](https://github.com/bazelbuild/rules_python/issues/178) to simply override *system* python with environment variable. I did look into *how* this could be implemented, but still being new to bazel I do not quite have the expertise to propose a change - tht was when I considered this less robust alternative.

Reply:
@fejta - is there a viable alternative to using `pip_import` that we can leverage here to dual-install two different versions of pylint?

Reply:
I think pip_import is a good practice we should probably continue using. 

I just want you to link to an issue so that if we see that its closed in 6 months, we know that we should try and have two versions (or maybe we won't use py2 by then). If it is still open we know to not bother.